conformance_tests/sysman/test_sysman_device/src/test_sysman_device_helper.cpp

```diff
@@ -12,45 +12,77 @@
+  properties.pNext = &ext_properties;
+  EXPECT_EQ(ZE_RESULT_SUCCESS,
+            zesDeviceGetProperties(sysman_device, &properties));
+  auto sysman_device_uuid = ext_properties.uuid;
+  sysman_device_uuids.push_back(TO_STD_ARRAY(sysman_device_uuid.id));
+}
+
+void get_sysman_sub_devices_uuids(zes_device_handle_t sysman_device,
+                                  std::vector<UUID> &sysman_device_uuids) {
+  uint32_t num_sub_devices = 0;
+  auto sub_device_properties =
+      lzt::get_sysman_subdevice_properties(sysman_device, num_sub_devices);
+  for (uint32_t sub_device_index = 0; sub_device_index < num_sub_devices;
+       sub_device_index++) {
+    sysman_device_uuids.push_back(
+        TO_STD_ARRAY(sub_device_properties[sub_device_index].uuid.id));
   }
 }
 
-void get_ze_root_uuids(std::vector<ze_device_handle_t> ze_devices,
-                       std::vector<ze_device_uuid_t> &ze_root_uuids,
-                       char *device_hierarchy) {
-  for (const auto &ze_device : ze_devices) {
-    ze_device_handle_t ze_root_device;
-    if (strcmp(device_hierarchy, "COMBINED") == 0) {
-      ze_root_device = lzt::get_root_device(ze_device);
-      if (ze_root_device == nullptr) {
-        ze_root_device = ze_device;
-      }
-    } else if (strcmp(device_hierarchy, "COMPOSITE") == 0) {
+UUID get_ze_device_uuid(ze_device_handle_t ze_device) {
+  auto ze_device_properties = lzt::get_device_properties(ze_device);
+  auto ze_device_uuid = ze_device_properties.uuid;
+  return TO_STD_ARRAY(ze_device_uuid.id);
+}
+
+void get_ze_root_uuid(ze_device_handle_t ze_device,
+                      std::vector<UUID> &ze_root_uuids,
+                      char *device_hierarchy) {
+  ze_device_handle_t ze_root_device;
+  if (strcmp(device_hierarchy, "COMBINED") == 0) {
+    ze_root_device = lzt::get_root_device(ze_device);
+    if (ze_root_device == nullptr) {
       ze_root_device = ze_device;
-    } else {
-      LOG_WARNING << "Unhandled ZE_FLAT_DEVICE_HIERARCHY mode:"
-                  << device_hierarchy;
-      continue;
-    }
-    auto root_device_properties = lzt::get_device_properties(ze_root_device);
-    auto root_uuid = root_device_properties.uuid;
-    if (std::find(ze_root_uuids.begin(), ze_root_uuids.end(), root_uuid) ==
-        ze_root_uuids.end()) {
-      ze_root_uuids.push_back(root_uuid);
     }
+  } else if (strcmp(device_hierarchy, "COMPOSITE") == 0) {
+    ze_root_device = ze_device;
+  } else {
+    LOG_WARNING << "Unhandled ZE_FLAT_DEVICE_HIERARCHY mode:"
+                << device_hierarchy;
+  }
+
+  auto root_uuid = get_ze_device_uuid(ze_root_device);
+  if (std::find(ze_root_uuids.begin(), ze_root_uuids.end(), root_uuid) ==
+      ze_root_uuids.end()) {
+    ze_root_uuids.push_back(root_uuid);
   }
 }
 
+void compare_core_and_sysman_uuid(std::vector<UUID> core_uuids,
+                                  std::vector<UUID> sysman_uuids) {
+  std::sort(core_uuids.begin(), core_uuids.end());
```

Comment:
Why is sorting needed here ?

Reply:
Not sure if the order of devices enumerated in core & sysman will be in same order. Hence before checking equality of UUID vector, sorting is done